service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt

```diff
@@ -60,6 +64,34 @@ class ChildAttendanceController(
         UserRole.MOBILE
     )
 
+    @GetMapping("/child/{childId}")
+    fun getChildSensitiveInfo(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable childId: UUID,
+        @RequestParam pin: String,
+        @RequestParam staffId: UUID
+    ): ResponseEntity<ChildResult> {
+        Audit.ChildSensitiveInfoRead.log(targetId = staffId)
+
+        db.read { it.handle.getEmployee(staffId) }?.let {
```

Comment:
- if the `staffId` is not found in the db, the check is silently passed so the pin check can be bypassed trivially by just passing a random UUID as staffId
- there are no role checks for the mobile device, so anybody (even citizens) can call this endpoint
- there are no role checks for the staff ID, so any employee can access information for any child
- one of the requirements was that audit logs show who is the actual employee accessing the data. This doesn't happen, because audit logs show the generic mobile device id

Reply:
Revised the whole implementation and added the staffId authorization